doc/cookbook/infinite-streams/InfiniteStreams.lhs

```diff
@@ -0,0 +1,277 @@
+# Serving infinite streams
+
+Servant provides facilities to work with streams of data. This is handy for cases where the data may take a while to
+fetch, but we can start returning data early. In this cookbook, we will be concerned with serving _infinite_ HTTP
+streams.
+
+HTTP streams have many advantages compared to other streaming standards like websockets: they are simple
+and are well-supported by a broad range of intermediaries, such as proxy servers, content-delivery networks, and
+corporate firewalls.
+
+An _infinite_ HTTP stream differs from a _finite_ stream in two major ways. First, cleaning up resources (such as
+connections) associated with infinite streams is not deterministic. We do not know when the handler will
+terminate -- if at all! Second, we want to ensure that the connection isn't cut off because no data is flowing.
+To keep the connection alive, we will need to send bytes on a regular basis.
+
+This is a Literate Haskell file, so let's get imports out of the way.
+
+```haskell
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeOperators #-}
+module Main (main) where
+
+-- from `aeson`
+import Data.Aeson (FromJSON, ToJSON)
+
+-- from `base`
+import Control.Concurrent (threadDelay, forkIO, killThread)
+import Control.Concurrent.Chan (Chan, newChan, readChan, writeChan)
+import Control.Concurrent.MVar (MVar, newEmptyMVar, putMVar, isEmptyMVar)
+import Control.Exception (throwIO, bracket)
+import Control.Monad (forever)
+import qualified Data.List
+import Data.Proxy (Proxy(Proxy))
+import GHC.Generics (Generic)
+
+-- from `http-client`
+import Network.HTTP.Client (newManager, defaultManagerSettings)
+
+-- from the `resourcet` package
+import Control.Monad.Trans.Resource (ReleaseKey)
+import Data.Acquire ( mkAcquire, Acquire )
+
+-- from `servant`
+import Servant
+    ( WithResource,
+      type (:>),
+      StreamGet,
+      NewlineFraming,
+      JSON,
+      SourceIO,
+      Context((:.), EmptyContext),
+      Handler,
+      Application )
+import qualified Servant.Types.SourceT as SourceT
+
+-- from `servant-client`
+import Servant.Client.Streaming (ClientM, mkClientEnv, client, withClientM, BaseUrl (..), Scheme (Http))
+
+-- from `servant-server`
+import Servant.Server (serveWithContext)
+
+-- from `warp`
+import qualified Network.Wai.Handler.Warp as Warp (run)
+```
+
+We start with our scenario: we are tasked with creating an API which will serve random numbers in real-time. We are
+given a function that creates a producer of characters, and a method for us to stop the producer:
```

Comment:
> a producer of characters

I thought we were producing an infinite stream of number. There does not seem to be any further mention of characters after that, have I missed something?